docs/fundamentals/code-analysis/style-rules/ide0049.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Use language keywords instead of framework type names for type references (IDE0049)"
+description: "Learn about code analysis rule IDE0049: Use language keywords instead of framework type names for type references"
+ms.date: 09/30/2020
+ms.topic: reference
+f1_keywords:
+- IDE0049
+- dotnet_style_predefined_type_for_locals_parameters_members
+- dotnet_style_predefined_type_for_member_access
+helpviewer_keywords:
+- IDE0049
+- dotnet_style_predefined_type_for_locals_parameters_members
+- dotnet_style_predefined_type_for_member_access
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+---
+# Use language keywords instead of framework type names for type references (IDE0049)
+
+|Property|Value|
+|-|-|
+| **Rule ID** | IDE0049 |
+| **Title** | Use language keywords instead of framework type names for type references |
+| **Category** | Style |
+| **Sub-category** | Language rules |
+| **Applicable languages** | C# and Visual Basic |
+
+## Overview
+
+This style rule can be applied to local variables, method parameters, and class members, or as a separate rule to type member access expressions. A value of **true** means prefer the language keyword (for example, `int` or `Integer`) instead of the type name (for example, `Int32`) for types that have a keyword to represent them. A value of **false** means prefer the type name instead of the language keyword.
```

Comment:
```suggestion
This style rule can be applied to local variables, method parameters, and class members, or as a separate rule to type-member access expressions. A value of **true** means prefer the language keyword (for example, `int` or `Integer`) instead of the type name (for example, `Int32`) for types that have a keyword to represent them. A value of **false** means prefer the type name instead of the language keyword.
```